### PR TITLE
minor cleanup of AspNetCore extension

### PR DESCRIPTION
**Pull request by brettsam**
minor cleanup of dependencies and dead code

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by jviau — APPROVED** [review]

**Review by fabiocav — APPROVED** [review]

**Review by liliankasem — APPROVED** [review]

File: extensions/Worker.Extensions.Http.AspNetCore/src/Worker.Extensions.Http.AspNetCore.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <!--Version information-->
     <VersionPrefix>2.0.0</VersionPrefix>
     <VersionSuffix>-preview4</VersionSuffix>
```
Comment (liliankasem) [inline extensions/Worker.Extensions.Http.AspNetCore/src/Worker.Extensions.Http.AspNetCore.csproj:10]: Do we need to remove preview today as well? Or is this for a different PR?

Comment (brettsam) [reply extensions/Worker.Extensions.Http.AspNetCore/src/Worker.Extensions.Http.AspNetCore.csproj:10]: it's a couple PRs away (we have to release the Worker/Worker.Core stuff first, then take that as a dependency here)

Comment (jviau) [inline extensions/Worker.Extensions.Http.AspNetCore/src/Worker.Extensions.Http.AspNetCore.csproj:10]: Do we want to remove version suffix in this PR or another?

Comment (liliankasem) [reply extensions/Worker.Extensions.Http.AspNetCore/src/Worker.Extensions.Http.AspNetCore.csproj:10]: I had the same comment here: https://github.com/Azure/azure-functions-dotnet-worker/pull/2854#discussion_r1838553922 :D

**Pull request merged**